overlord/snapstate/snapmgr.go

```diff
@@ -76,7 +76,8 @@ type SnapSetup struct {
 
 	Flags SnapSetupFlags `json:"flags,omitempty"`
 
-	SnapPath string `json:"snap-path,omitempty"`
+	SnapPath string        `json:"snap-path,omitempty"`
+	SideInfo snap.SideInfo `json:"side-info,omitempty"`
```

Comment:
I think it's the right path here, otoh this replicates info we have also differently in the setup (revision for example as shown by the issue later) :/, I think it would be safer to have a pointer here : *snap.SideInfo